basiclibrary/src/main/java/basiclibrary/Library.java

```diff
@@ -61,4 +59,59 @@ public String calAvg(int[] inputArr){
         }
         return inputArrs[idx];
     }
+
+
+
+    //Lab03 Method#1: Analyzing Weather Data
+    public HashMap<String,Integer> analyzeWeatherData(int[][] octSeattleWeather){
+        int min =999;
+        int max =0;
+        //create hashmap to store the final result want to return
+        HashMap<String,Integer> res = new HashMap<String,Integer>();
+        //create a hashset to store all the numbers appears in the input weather data
+        HashSet<Integer> set = new HashSet<Integer>();
+        //iterate the input weather data(matrix) find the lowest temperature and highest temperture and store the unique temperature in the set
+
+        for(int i=0;i<octSeattleWeather.length;i++){
+            for(int j=0;j<octSeattleWeather[i].length;j++){
+                min = min<octSeattleWeather[i][j]?min:octSeattleWeather[i][j];
+                max = max>octSeattleWeather[i][j]?max:octSeattleWeather[i][j];
+                set.add(octSeattleWeather[i][j]);
+            }
+        }
+        //insert "High" and "Low" to the Hashmap
+        res.put("High",max);
+        res.put("Low",min);
+       //between High and Min,if set doesn't have certain number, add that to the hashmap ,the key will be ("Never Saw temperature"+K) cuz the key has to be unique that is why i added k after that string,but when print out i will take that "k" out, so you will only see "Never saw temperature"
+        for(int k=0;k<(max-min);k++){
+           if(!set.contains(k+min)) {
+               res.put("Never saw temperature"+k,(k+min));
```

Comment:
Instead of having to go through the HashMap again to print to the console, inside this loop you could have just `System.out.println("Never saw temperature " + (k + min)`

Reply:
Hi, i did it on purpose, because we  need to do the test, if you print out right away, it is not benefitial for the junit testing later on.

Reply:
Fair enough 👍